user-interface/1004-ASSO-associate.md

```diff
@@ -1 +1,69 @@
-# Associate and disassociate governance tokens with a Vega key
\ No newline at end of file
+# Associate and disassociate governance tokens with a Vega key
+
+The Governance token on a Vega network is an ERC20 ethereum token. It has two utilities on Vega,
+
+- Staking the Proof of stake network,
+- Participating in Governance.
+
+To use the Governance token on a Vega network it first needs to be associated with a Vega key/party. This Vega key can then stake, propose and vote.
+
+The word "associate" is used in some user interfaces, as apposed the word "stake" in function names. Stake can be avoided to prevent users thinking they would get a return only after the staking step. On Vega `Staking = Association + Nomination`, as in you need to run the "stake " function on Ethereum but then the nominate step on Vega before you get staking income. See [Glossary](../glossaries/staking-and-governance.md).
+
+Associated tokens also count as vote weight in on-Vega governance (new markets etc), A parties vote weight is backed by the number of Governance tokens associated with that Vega key/party.
+
+Associating tokens to a Vega key is a little like depositing, except, a deposit can only be released by the Vega network, where as an association can be revoked on ethereum by the the eth key that did the association.
+
+Governance tokens may be held by a [Vesting contract](1005-VEST-vesting.md).
+
+## Token discovery
+
+When looking to acquire governance tokens, I...
+
+- **must** see the contract address for the governance token of the Vega network <a name="1004-ASSO-001" href="#1004-ASSO-001">1004-ASSO-001</a>
+
+...so I can participate in governance and staking.
+
+## Associate
+
+When looking to stake validators or participate in governance, I first need to associate governance tokens with a Vega wallet/key, I...
+
+- **must** [connect an Ethereum wallet/key](0004-EWAL-connect_ethereum_wallet.md) to see tokens it may have in wallet or attributed to it in the vesting contract <a name="1004-ASSO-002" href="#1004-ASSO-002">1004-ASSO-002</a>
+- **must** select a Vega key to associate to <a name="1004-ASSO-003" href="#1004-ASSO-003">1004-ASSO-003</a>
+  - **must** be able use a [connected Vega wallet](0002-WCON-connect_vega_wallet.md) as instead of manually inputting a public key <a name="1004-ASSO-004" href="#1004-ASSO-004">1004-ASSO-004</a>
+  - **should** be able to populate field with a string, so I can associate to a wallet I without connecting it
+- if the connected ethereum wallet has vesting tokens: **must** be able to select to associate from either the vesting contract or the wallet <a name="1004-ASSO-006" href="#1004-ASSO-006">1004-ASSO-006</a>
```

Comment:
question: should we add something here for the case where I only have one of vesting or wallet?

Reply:
added some notes comments in https://github.com/vegaprotocol/specs/pull/1301/commits/2407b1f610407595d3d393f26a30e24f2478c7ba 
> - if the connected ethereum wallet has vesting tokens: **must** be able to select to associate from either the vesting contract or the wallet (The default should be wallet and the option to use vesting tokens should only appear if there are tokens in a tranche (associated or not)) <a name="1004-ASSO-006" href="#1004-ASSO-006">1004-ASSO-006</a>

Does that work? or are you thinking we should have another AC code so separate tests can reference?

Reply:
also added some codes to check the values in your eth wallet are updated: https://github.com/vegaprotocol/specs/pull/1301/commits/db1d17adb01d0fa941d3fb4242d9463706e0edbb